addons/snailmail/models/snailmail_letter.py

```diff
@@ -414,7 +422,9 @@ def _format_snailmail_failures(self):
         return failures_infos
 
     def _append_cover_page(self, invoice_bin: bytes):
-        address = self.partner_id.with_context(show_address=True, lang='en_US')._get_name().replace('\n', '<br/>')
+        address_split = self.partner_id.with_context(show_address=True, lang='en_US')._get_name().split('\n')
+        address_split[0] = self.partner_id.name or self.partner_id.parent_id and self.partner_id.parent_id.name or address_split[0]
```

Comment:
```suggestion
        address_split[0] = address_split[0] or self.partner_id.parent_id and self.partner_id.parent_id.name
```

Reply:
As discussed on Discord, we want to use the auto-generated recipient name as a last resort if we cannot find an implicit name for the partner. Hence why `address_split[0]` is last in the list.

e.g.: We have the partner "Deco Addict" and we create an invoice address for it. This new invoice address will be a Partner categorized as an invoice address and its name will be `None`.

When we use `_get_name()`, the function understands it's an invoice address and thus auto-fills the name as "Deco Addict, Invoice address", but this is quite ugly to send on a letter as a recipient name.
That's why I first check to see if the user input a name manually for the Invoice address, then if they didn't I check the original partner (Deco Addict in this case), and if we somehow don't have a name or a parent Partner, I default to the auto-generated name !